DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -857,6 +857,12 @@ public DMASTProcStatement ProcStatement()
 
                             initializer = new DMASTProcStatementExpression(variable.Location, new DMASTAssign(variable.Location, variable, value));
                         }
+                    } else {
```

Comment:
```suggestion
                    } else if(Current().Type != TokenType.DM_Comma && Current().Type != TokenType.DM_Semicolon) {
```
Needed otherwise loops without initializers like `for (,user.bhunger>0,user.bhunger--)` will break

Reply:
Not ideal but for loops need a rewrite anyways.